src/v/cluster/topics_frontend.cc

```diff
@@ -689,9 +689,34 @@ ss::future<std::vector<topic_result>> topics_frontend::autocreate_topics(
     }
     // current node is a leader controller
     if (leader == _self) {
+        // NOTE since we are skipping the preprocessing done in
+        // create_topics_handler::handle, it is partially replicated here
+        auto to_create = without_custom_assignments(std::move(topics));
+        auto& cfg = config::shard_local_cfg();
+        for (auto& tp : to_create) {
+            tp.cfg.properties.shadow_indexing = [&] {
+                auto read = cfg.cloud_storage_enable_remote_read();
+                auto write = cfg.cloud_storage_enable_remote_write();
+
+                if (read && write) {
+                    return model::shadow_indexing_mode::full;
+                }
+                if (read) {
+                    return model::shadow_indexing_mode::fetch;
+                }
+                if (write) {
+                    return model::shadow_indexing_mode::archival;
+                }
+                return model::shadow_indexing_mode::disabled;
+            }();
+            if (!tp.cfg.properties.cleanup_policy_bitflags.has_value()) {
+                tp.cfg.properties.cleanup_policy_bitflags
+                  = cfg.log_cleanup_policy();
+            }
```

Comment:
My reading of the `kafka/server` side suggest that it's just the cleanup policy and shadow indexing setting that might diverge from client provided properties. Does that match your understanding? Were there other properties you were thinking about setting up here?

Reply:
yes, based on `topic_autocreate_test` only `shadow_indexing` and `cleanup_policy_bitflags` diverge. To write these lines, I skimmed through https://github.com/redpanda-data/redpanda/blob/6e70513d845094c5419765c19968ebfbbeed555b/src/v/kafka/server/handlers/topics/types.cc#L143-L219 to see what other property was missing for an auto-created topic, but it seems only these two matters.
But the test itself is quite simple, and I didn't spend too much time on it, so I can't exclude that in a different context some other property can end up missing.